shell/platform/android/io/flutter/embedding/engine/android/FlutterView.java

```diff
@@ -0,0 +1,95 @@
+import android.util.Log;
+import android.widget.FrameLayout;
+
+/**
+ * Displays a Flutter UI on an Android device.
+ * <p>
+ * A {@code FlutterView}'s UI is painted by a corresponding {@link FlutterEngine}.
+ * <p>
+ * A {@code FlutterView} can operate in 2 different {@link RenderMode}s:
+ * <ol>
+ *   <li>{@link RenderMode#surface}, which paints a Flutter UI to a {@link android.view.SurfaceView}.
+ *   This mode has the best performance, but a {@code FlutterView} in this mode cannot be positioned
+ *   between 2 other Android {@code View}s in the z-index, nor can it be animated/transformed.</li>
+ *   <li>{@link RenderMode#texture}, which paints a Flutter UI to a {@link android.graphics.SurfaceTexture}.
+ *   This mode is not as performant as {@link RenderMode#surface}, but a {@code FlutterView} in this
+ *   mode can be animated and transformed, as well as positioned in the z-index between 2+ other
+ *   Android {@code Views}.</li>
+ * </ol>
+ */
+public class FlutterView extends FrameLayout {
+  private static final String TAG = "FlutterView";
+
+  @NonNull
+  private RenderMode renderMode;
+
+  /**
+   * Constructs a {@code FlutterSurfaceView} programmatically, without any XML attributes.
+   *
+   * {@link #renderMode} defaults to {@link RenderMode#surface}.
+   */
+  public FlutterView(@NonNull Context context) {
+    this(context, null, null);
+  }
+
+  /**
+   * Constructs a {@code FlutterSurfaceView} programmatically, without any XML attributes,
+   * and allows selection of a {@link #renderMode}.
+   */
+  public FlutterView(@NonNull Context context, @NonNull RenderMode renderMode) {
+    this(context, null, renderMode);
+  }
+
+  /**
+   * Constructs a {@code FlutterSurfaceView} in an XML-inflation-compliant manner.
+   *
+   * // TODO(mattcarroll): expose renderMode in XML when build system supports R.attr
+   */
+  public FlutterView(@NonNull Context context, @Nullable AttributeSet attrs) {
+    this(context, attrs, null);
+  }
+
+  private FlutterView(@NonNull Context context, @Nullable AttributeSet attrs, @Nullable RenderMode renderMode) {
+    super(context, attrs);
+
+    this.renderMode = renderMode == null ? RenderMode.surface : renderMode;
+
+    init();
+  }
+
+  private void init() {
+    Log.d(TAG, "Initializing FlutterView");
+
+    switch (renderMode) {
+      case surface:
+        Log.d(TAG, "Internally creating a FlutterSurfaceView.");
+        FlutterSurfaceView flutterSurfaceView = new FlutterSurfaceView(getContext());
+        addView(flutterSurfaceView);
+        break;
+      case texture:
+        Log.d(TAG, "Internally creating a FlutterTextureView.");
+        FlutterTextureView flutterTextureView = new FlutterTextureView(getContext());
+        addView(flutterTextureView);
+        break;
+    }
+  }
+
+  /**
+   * Render modes for a {@link FlutterView}. See {@link FlutterView} for more information
+   * about how render modes compare.
+   */
+  public enum RenderMode {
+    surface,
```

Comment:
nit: These also need Javadoc.

https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required

Reply:
What would you like them to say?  I can't think of anything that isn't superfluous.  The JavaDoc for the surrounding class already provides deep discussion of what these are.

Reply:
If you'd rather not write it twice I think it would make more sense to define what these are in depth on the members here rather than in the summary of `FlutterView`, and then have `FlutterView` shortened and say to look at the docs on these enums for more details if needed. Somebody who sees these members referenced in code and jumps to them in their IDE to get more info on them isn't necessarily going to be able to intuit that they need to look at `FlutterView`'s class summary to get a definition of each one.

Reply:
I went ahead and copied the explanations down.  I don't think we want to force developers who are reading the class doc in a javadoc site to jump to another page to figure out when to use these modes.

Reply:
This is good, just a small nit. The straight copy paste makes this a little hard to read. You ideally want a single summary fragment to start this off, similar to dart docs.

https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment